modules/reference/pages/command-line-tools.adoc

```diff
@@ -695,6 +695,68 @@ Show help message and exit.
 
 
 
+[[spacewalk-repo-sync]]
+== Synchronize yum Repositories into {productname} Channels (spacewalk-repo-sync)
+
+[command]``spacewalk-repo-sync`` is an important tool on {productname}, which works most of the time in the background.
+With [command]``spacewalk-repo-sync`` you mainly synchronize yum repositories into {productname} channels.
+
+But for debugging or solving smaller issues it is useful to run it directly with special parameters.
+
+
+
+=== Normal Channel Synchronization
+
+Basic operation:
+
+[command]``spacewalk-repo-sync --list``::
+List all custom channels and the repositories assigned to them.
+[command]``spacewalk-repo-sync  --channel <custom-channel>``::
+Synchronize a single channel ``<custom-channel>`` to all repositories assigned to it with the {webui} or the API.
+
+
+
+=== Solve Checksum Problems
+
+Use the [option]``--deep-verify`` option to ignore cached package checksums, and thus solving checksum problems.
+
+
+
+=== Force Re-import Patches (Errata)
+
+Use the [option]``--force-all-errata`` to force re-importing all the patches (errata).
+To speed it up you can combine it with the [option]``--no-packages`` option.
+
+To find the root cause of synchronization problems it often helps to view the HTTP log when command]``spacewalk-repo-sync`` is running.
+
+. Set and export the [systemitem]``URLGRABBER_DEBUG``:
++
+----
+export URLGRABBER_DEBUG=DEBUG
```

Comment:
Hmm, I fear after the usage of "zypper" this might not work anymore or will give an incomplete result.

@meaksh How would we debug today? The goal is to get the full HTTP protocol conversation printed to the logs.

For zypper you can set the following environment variable:

    ZYPP_MEDIA_CURL_DEBUG=<1|2> Log http headers, if 2 also log server responses.

But we do not use zypper exclusive. For the Downloader part would the URLGRABBER_DEBUG option still work?
Also with python3 ?

Reply:
There are two things here:

Setting `ZYPP_MEDIA_CURL_DEBUG=<1|2>` as env variable would work for the metadata downloading part made by Zypper. So, i.a., `ZYPP_MEDIA_CURL_DEBUG=2 spacewalk-repo-sync -c MYCHANNEL` will log the HTTP conversation into `/var/log/zypper.log`.

For the RPM downloading part, we would need to keep using `URLGRABBER_DEBUG` (since the backend for downloading the RPM is still `urlgrabber`), but I just tested it now and it seems there is an issue when running on Python3.

I've just filed a bug for this: https://bugzilla.suse.com/show_bug.cgi?id=1152514

Reply:
@meaksh thanks for checking and feedback!